crates/router/src/core/payment_methods/cards.rs

```diff
@@ -2193,3 +2237,42 @@ pub async fn delete_payment_method(
         },
     ))
 }
+
+pub async fn create_encrypted_payment_method_data(
+    state: &routes::AppState,
+    merchant_account: &domain::MerchantAccount,
+    pm_data: Option<PaymentMethodsData>,
+) -> Option<Encryption> {
+    let db = &*state.store;
+    let key_store = db
+        .get_merchant_key_store_by_merchant_id(
+            &merchant_account.merchant_id,
+            &db.get_master_key().to_vec().into(),
+        )
+        .await
+        .ok();
```

Comment:
We have the key store already available in all caller contexts, so we can just pass it in as an argument to this function.

Reply:
have done the corresponding changes